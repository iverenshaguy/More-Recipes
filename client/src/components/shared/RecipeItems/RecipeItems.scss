@import '../../../assets/scss/modules';

#search-result-wrapper {
  hr {
    border-top-color: $border-top-grey;
    margin-top: 2rem;
    width: 30%;
  }
}

#search-result {
  #side-sorter {
    border-right: 1px solid $light-grey;
  }
}
