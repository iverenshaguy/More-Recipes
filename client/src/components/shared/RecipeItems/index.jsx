import React from 'react';
import PropTypes from 'prop-types';
import RecipeCard from '../RecipeCard';
import Pagination from '../Pagination';
import getNoResultTitle from '../../../helpers/getNoResultTitle';
import './RecipeItems.scss';

/**
 * @exports
 * @function RecipeItems
 * @param {object} props
 * @returns {component} RecipeItems
 */
const RecipeItems = props => (
  <div className="container-fluid pb-5 background-grey page-margin" id="search-result-wrapper">
    <div className="heading pt-4 text-center" id="search-result-heading">
      <div className="col-xs-12 col-md-12 pt-3" id="title">
        <h5>{props.title}</h5>
      </div>
    </div>
    <hr />
    <div className="row pt-2 px-5" id="search-result">
      {props.recipes.length === 0 &&
        <div className="col">
          {props.recipes.length === 0 && <div className="text-center pt-4">{getNoResultTitle(props.title)}</div>}
        </div>}
      {props.recipes.length !== 0 &&
        props.recipes.map(recipe => (<RecipeCard key={recipe.id} recipe={recipe} />))}
    </div>
    <Pagination
      items={props.recipes}
      onPageChange={props.handlePageChange}
      metaData={props.metaData}
    />
  </div>);

RecipeItems.propTypes = {
  title: PropTypes.string.isRequired,
  recipes: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    recipeName: PropTypes.string,
    recipeImage: PropTypes.string,
    prepTime: PropTypes.string,
    cookTime: PropTypes.string,
    totalTime: PropTypes.string,
    difficulty: PropTypes.string,
    extraInfo: PropTypes.string,
    vegetarian: PropTypes.bool,
    ingredients: PropTypes.array,
    preparations: PropTypes.array,
    directions: PropTypes.array,
    upvotes: PropTypes.number,
    downvotes: PropTypes.number,
    views: PropTypes.number,
    createdAt: PropTypes.string,
    updatedAt: PropTypes.string,
    userId: PropTypes.number,
    rating: PropTypes.string
  })).isRequired,
  metaData: PropTypes.shape({
    firstPage: PropTypes.number,
    lastPage: PropTypes.number,
    page: PropTypes.number,
    pageRecipeCount: PropTypes.number,
    pages: PropTypes.arrayOf(PropTypes.number),
    totalRecipeCount: PropTypes.number,
  }).isRequired,
  handlePageChange: PropTypes.func.isRequired
};

export default RecipeItems;
