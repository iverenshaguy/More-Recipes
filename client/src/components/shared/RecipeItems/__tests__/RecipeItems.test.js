import React from 'react';
import RecipeItems from '../index';

describe('RecipeItems', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RecipeItems />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
