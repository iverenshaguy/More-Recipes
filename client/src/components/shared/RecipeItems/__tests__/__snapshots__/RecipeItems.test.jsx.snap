// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeItems renders correctly 1`] = `
<div
  className="container-fluid pb-5 background-grey page-margin"
  id="search-result-wrapper"
>
  <div
    className="heading pt-4 text-center"
    id="search-result-heading"
  >
    <div
      className="col-xs-12 col-md-12 pt-3"
      id="title"
    >
      <h5>
        TOP RECIPES
      </h5>
    </div>
  </div>
  <hr />
  <div
    className="row pt-2 px-5"
    id="search-result"
  >
    <RecipeCard />
    <RecipeCard />
    <RecipeCard />
    <RecipeCard />
  </div>
</div>
`;
